packages/neon-core/src/tx/transaction/main.ts

```diff
@@ -23,51 +35,87 @@ export function serializeArrayOf(prop: any[]): string {
   return num2VarInt(prop.length) + prop.map(p => p.serialize()).join("");
 }
 
-export function deserializeType(
+export function deserializeVersion(
   ss: StringStream,
   tx: Partial<TransactionLike> = {}
 ): Partial<TransactionLike> {
   const byte = ss.read();
-  return Object.assign(tx, { type: parseInt(byte, 16) });
+  const version = parseInt(byte, 16);
+  if (version !== 0) {
+    log.warn(`Transaction version should be 0 not ${version}`);

```

Comment:
why warning here? I believe the version determines how we deserialize this so if we are encountering an unknown version we should not be able to deserialize this correctly and throw an error.

Reply:
Changed